lv2 + sratom + lilv: support cross-compilation

Mindavi opened the pull request:
###### Motivation for this change

Support cross-compilation for these packages.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux (output tree is equal to old tree)
  - [x] aarch64-linux (cross-compiled)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Mindavi committed:
sratom: support cross-compilation

Mindavi committed:
lilv: support cross-compilation

Artturin commented on pkgs/development/libraries/audio/lilv/default.nix:
```diff
@@ -14,6 +14,11 @@ stdenv.mkDerivation rec {
+  dontAddWafCrossFlags = true;
+
+  preConfigure = ''
+    export PKGCONFIG="$PKG_CONFIG"
```
since waf uses the `PKGCONFIG` variable
https://gitlab.com/ita1024/waf/-/blob/master/waflib/Tools/c_config.py#L213

maybe we should set it in the `wafConfigurePhase` https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/build-managers/wafHook/setup-hook.sh

something like
```bash
if [ -n "${PKG_CONFIG}" ]; then
  export PKGCONFIG="${PKG_CONFIG}"
fi

Mindavi replied:
Makes sense to me. I'll try that.

Mindavi replied:
Thanks again for the suggestion, I applied it and it seems to work well.

Mindavi committed:
wafHook: export PKGCONFIG if needed

This is useful for cross-compilation, where nixpkgs sets PKG_CONFIG.
waf decided to use PKGCONFIG instead, so we'll set it when necessary.

Mindavi committed:
jackaudio: remove unneeded PKGCONFIG variable

Not needed anymore after setting this correctly in the wafHook.

Mindavi committed:
lv2: support cross-compilation

The waf cross flags seem to not be useful for most packages, to me.
I'll just set the flag for now, but maybe turning it around
may be more useful to do somewhere in the future.

SuperSandro2000 commented:
Maybe we should think about disabling waf cross flags by default if they are not that useful.

Mindavi commented:
Yeah, thought about that already and I think it'd be a good idea. It'd require some documentation updates and testing of the packages that do require it (they exist). Depending on how many packages need it it may even be more logical to put them in the derivations itself.

It's also unfortunate that the flags introduce qemu to all packages that use it, which can cause a native rebuild of qemu before the package actually builds.

Mindavi commented:
Will look into converting from dontUseWafCross flags, but I'll have to see where that goes.
